doc/protocol.md

```diff
@@ -829,6 +829,119 @@ or pipe is done.
 Additionally, a "options" control message may be sent in this channel
 to change the "max-size" options.
 
+Payload: fsinfo
+----------------
+
+This channel is used to report information about a given path, and the file or
+a directory which may reside there, including the possibility of updates.  In
+the case of directories, information can also be reported about the (directly)
+contained entries.
+
+Additional "open" command options should be specified with a channel of
+this payload type:
+
+ * `path` (mandatory): a string
+ * `attrs` (mandatory): a list of strings, the list of attributes to be
+   reported for the requested file.  See below for the possibilities.
+ * `fnmatch`: (optional, default ''): if non-empty, this is a [`fnmatch`-style
+   pattern](https://docs.python.org/3/library/fnmatch.html) selecting the directory
+   entries to be reported.  Use `*` to report all files.  Can also be used to
+   implement search (`searc*`) or hide temporary files (`[!.]*`).
+ * `close-on-error` (optional, default: true): if the channel should close
+   (with `problem`) on failure to `open()` the named `path`
+ * `watch` (optional, default: false): if the channel should watch for changes
+ * `follow` (default: true): if the trailing component of `path` is a symbolic
+   link, whether we should follow it.  `no-follow` mode is not supported for
+   `watch` (but is also mostly unnecessary).
+
+A channel is created with an associated absolute pathname ('path').  A path is
+a string that you pass to the `open()` syscall.  The object returned by that
+call is described by the (JSON Document) value of the channel.
+
+Strictly speaking, the path is specified to the channel as a unicode string,
+but paths on the filesystem are binary strings.  This mapping is handled using
+the usual Python convention: filenames are stored in utf8 and encoded and
+decoded with `errors='surrogateescape'`, which allows for handling of filenames
+which are not utf8.
+
+Conceptually, the channel represents the value of a JSON object describing what
+is found at the path, or the reason for an error.  Each message on the channel
+is a JSON Merge Patch (RFC7396) which updates or replaces this object.
```

Comment:
Link to https://datatracker.ietf.org/doc/html/rfc7396 ?

Reply:
OK.